ui/app/components/modals/cancel-transaction/cancel-transaction.container.js

```diff
@@ -0,0 +1,55 @@
+import { connect } from 'react-redux'
+import { compose } from 'recompose'
+import R from 'ramda'
+import { multiplyCurrencies } from '../../../conversion-util'
+import { bnToHex } from '../../../helpers/conversions.util'
+import withModalProps from '../../../higher-order-components/with-modal-props'
+import CancelTransaction from './cancel-transaction.component'
+import { showModal, hideModal, createCancelTransaction } from '../../../actions'
+
+const mapStateToProps = (state, ownProps) => {
+  const { metamask } = state
+  const { transactionId, originalGasPrice } = ownProps
```

Comment:
Is there any way we can structure the data from `withModalProps` so that it is clear where `transactionId` and `originalGasPrice` come from?

Reply:
Did you consider just adding a selector for modal props, instead of a higher order component? One down side of the higher order component is that it adds another potential source of data to a component, in addition to state and just a simple pass from the parent.